### ISSUE TITLE
What is ngrok used for in local deployment

### ISSUE THREAD

**Issue by angerthosenear** [carefree0910/carefree-creator#13]
Another FAQ item for you.

What's the use of ngrok in the local deployment steps? If it's running locally why do we need to setup ngrok? Or is it solely for sharing the local setup with others with the url?

If it's only for sharing with others, I feel like local deploy and ngrok/bore setup/running could be separate optional steps.

**Comment by carefree0910** [comment]
Hi!

If I'm not mistaken, I actually did not mention `ngrok` in the `README`, and it is only used in the Google Colab server, because we need to access Colab via `ngrok`. 😉

I'm not sure if this answers your question, feel free to comment more if you are still confused! 😆

**Comment by phi-line** [comment]
> What's the use of ngrok in the local deployment steps

@angerthosenear [ngrok](https://ngrok.com/) is what is known as a "reverse proxy" and allows for a local port to become exposed to the open internet! It's really useful for these types of connections between local/external services, since it removes some of the pain of setting up ssh tunnels / port forwarding / web servers, and the like.

**Comment by angerthosenear** [comment]
Oh interesting. Last time I clicked the link in the readme page for local installation, it brought me to no where. So I used the one that was labelled as recommended. Which is obviously for the cloud one. I tried trimming out the ngrok things to get it going but that didn't quite work.

Tried again today and the links were working correctly getting me to the right steps for local setup.

And, I've used ngrok and bore quite a bit so I'm familiar with them. That's why I was confused to see them.
